### PR TITLE
strip prefix 'v' in coredns version if exists

### LINKED ISSUE #100344: ServiceAccount admission controller migration [Feature:BoundServiceAccountTokenVolume] master upgrade should maintain a functioning cluster
#### Which jobs are failing:

sig-auth-serviceaccount-admission-controller-migration

#### Which test(s) are failing:

`ServiceAccount admission controller migration [Feature:BoundServiceAccountTokenVolume] master upgrade should maintain a functioning cluster`

#### Since when has it been failing:

3/9

#### Testgrid link:

https://testgrid.k8s.io/sig-auth-gce#upgrade-tests&width=20

#### Reason for failure:

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/cloud/gcp/auth/service_account_admission_controller_migration.go:37 You may only call BeforeEach from within a Describe, Context or When /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:185
```

#### Anything else we need to know:

/sig auth
/assign @zshihang

### PR DESCRIPTION
**Pull request by zshihang**
#### What type of PR is this?

/kind bug
/kind failing-test

#### Which issue(s) this PR fixes:

Fixes #100344

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/upgrade.sh [hunk 1]
```diff
@@ -552,7 +552,7 @@ function update-coredns-config() {
 }
 
 echo "Fetching the previously installed CoreDNS version"
-CURRENT_COREDNS_VERSION=$("${KUBE_ROOT}/cluster/kubectl.sh" -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | cut -d ":" -f 2)
+CURRENT_COREDNS_VERSION=$("${KUBE_ROOT}/cluster/kubectl.sh" -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | sed -r 's/.+:v?(.+)/\1/'
```
Comment (jeremyje) [inline cluster/gce/upgrade.sh:555]: I think it would be more resilient if the cut command stayed and optionally chopping off the v prefix.

**Review by jeremyje — CHANGES_REQUESTED** [review]

**Comment by liggitt** [comment]
/assign @bowei @MrHohn 

it looks like the recent coredns bump broke upgrade scripts/tests. Can you review this to see if this makes sense?

File: cluster/gce/upgrade.sh [hunk 2]
```diff
@@ -480,7 +480,7 @@ function update-coredns-config() {
   fi
 
   echo "== Fetching the latest installed CoreDNS version =="
-  NEW_COREDNS_VERSION=$("${KUBE_ROOT}"/cluster/kubectl.sh -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | cut -d ":" -f 2)
+  NEW_COREDNS_VERSION=$("${KUBE_ROOT}"/cluster/kubectl.sh -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | sed -r 's/.+:v?(.+)/\1/'
```
Comment (pacoxu) [inline cluster/gce/upgrade.sh:483]: missing `)` at the end

File: cluster/gce/upgrade.sh (hunk 1 above)
Comment (pacoxu) [reply cluster/gce/upgrade.sh:555]: same with above

Comment (rajansandeep) [reply cluster/gce/upgrade.sh:555]: The current version would be without the v prefix (k8s.gcr.io/coredns:1.7.0)and the new version would be with the v prefix (k8s.gcr.io/coredns/coredns:v1.8.0). I think sed would cover both cases.

File: cluster/gce/upgrade.sh (hunk 2 above)
Comment (zshihang) [reply cluster/gce/upgrade.sh]: good catch. done

**Pull request merged**